### PR TITLE
Fix: Properly handle backup creation and deletion errors

### LINKED ISSUE #806: Deletion of backups fails if T4C server is not reachable
Backups can not be deleted if the T4C server is not reachable:

```
time="2023-06-27 17:52:44,362" level=ERROR function=dispatch client="10.128.67.53:42858" trace_id="VP93P0Q3DV" method="DELETE" path="/api/v1/projects/gse/models/gse/backups/pipelines/81" user="moritzweber" message="HTTPSConnectionPool(host='t4c.ssi.dbmc.comp.db.de', port=443): Max retries exceeded with url: /api/v1.0/users/techuser-1bcd1ccf-bcf1-4a40-bb9c-1082d6682baa?repositoryName=GSE (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fc38fab9d50>: Failed to establish a new connection: [Errno -2] Name or service not known'))"
Traceback (most recent call last):
File "/tmp/backend/capellacollab/core/logging.py", line 75, in dispatch
return await call_next(request)
^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 84, in call_next
raise app_exc
File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 70, in coro
await self.app(scope, receive_or_disconnect, send_no_error)
File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 108, in __call__
response = await self.dispatch_func(request, call_next)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/tmp/backend/capellacollab/core/logging.py", line 86, in dispatch
response: fastapi.Response = await call_next(request)
^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 84, in call_next
raise app_exc
File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 70, in coro
await self.app(scope, receive_or_disconnect, send_no_error)
File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 108, in __call__
response = await self.dispatch_func(request, call_next)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/starlette_prometheus/middleware.py", line 57, in dispatch
raise e from None
File "/usr/local/lib/python3.11/site-packages/starlette_prometheus/middleware.py", line 53, in dispatch
response = await call_next(request)
^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 84, in call_next
raise app_exc
File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 70, in coro
await self.app(scope, receive_or_disconnect, send_no_error)
File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
raise exc
File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
await self.app(scope, receive, sender)
File "/usr/local/lib/python3.11/contextlib.py", line 222, in __aexit__
await self.gen.athrow(typ, value, traceback)
File "/usr/local/lib/python3.11/site-packages/fastapi/concurrency.py", line 36, in contextmanager_in_threadpool
raise e
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='', port=443): Max retries exceeded with url: /api/v1.0/users/techuser-1bcd1ccf-bcf1-4a40-bb9c-1082d6682baa (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fc38fab9d50>: Failed to establish a new connection: [Errno -2] Name or service not known'))
```

### PR DESCRIPTION
**Pull request by dominik003**
Resolves #806

### CODE REVIEW

File: frontend/src/app/projects/models/backup-settings/trigger-pipeline/trigger-pipeline.component.html [hunk 1]
```diff
@@ -106,6 +106,14 @@ <h3>Additional actions</h3>
         >
           View runs
         </a>
+        <mat-slide-toggle
+          *ngIf="userService.validateUserRole('administrator')"
+          [(ngModel)]="force"
+          aria-label="Toggle button"
```
Comment (MoritzWeber0) [inline frontend/src/app/projects/models/backup-settings/trigger-pipeline/trigger-pipeline.component.html:112]:
```suggestion
          aria-label="Force deletion of pipeline"
```

File: frontend/src/app/projects/models/backup-settings/trigger-pipeline/trigger-pipeline.component.html [hunk 2]
```diff
@@ -106,6 +106,14 @@ <h3>Additional actions</h3>
...
+          matTooltip="Delete pipeline even if t4c instance not unreachable"
```
Comment (MoritzWeber0) [inline frontend/src/app/projects/models/backup-settings/trigger-pipeline/trigger-pipeline.component.html:113]:
```suggestion
          matTooltip="Delete pipeline even if T4C instance not reachable"
```

File: frontend/src/app/projects/models/backup-settings/trigger-pipeline/trigger-pipeline.component.html [hunk 3]
```diff
@@ -106,6 +106,14 @@ <h3>Additional actions</h3>
...
+        >
+          Force Deletion
```
Comment (MoritzWeber0) [inline frontend/src/app/projects/models/backup-settings/trigger-pipeline/trigger-pipeline.component.html:115]:
```suggestion
          Force deletion
```

File: backend/capellacollab/tools/exceptions.py [hunk 4]
```diff
@@ -24,7 +24,30 @@ async def tool_version_not_found_exception_handler(
...
+
+async def tool_image_not_found_exception_handler(
+    request: fastapi.Request, exc: ToolImageNotFoundError
+) -> fastapi.Response:
+    return await exception_handlers.http_exception_handler(
+        request,
+        fastapi.HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail={
+                "reason": f"The tool with id {exc.tool_id} does not have an {exc.image_name} image."
```
Comment (MoritzWeber0) [inline backend/capellacollab/tools/exceptions.py:41]:
```suggestion
                "reason": f"The tool with id {exc.tool_id} doesn't have an {exc.image_name} image."
```

**Review by MoritzWeber0 — APPROVED** [review]
Just some small comments.

File: frontend/src/app/projects/models/backup-settings/trigger-pipeline/trigger-pipeline.component.ts [hunk 5]
```diff
@@ -45,10 +48,13 @@ export class TriggerPipelineComponent implements OnInit {
     public pipelineService: PipelineService,
     private pipelineRunService: PipelineRunService,
     public sessionService: SessionService,
+    public userService: UserService,
     private router: Router
   ) {}
 
   ngOnInit(): void {
+    this.userService.updateOwnUser();
```
[the remark this answers is not in the document]
Comment (dominik003) [reply frontend/src/app/projects/models/backup-settings/trigger-pipeline/trigger-pipeline.component.ts:56]: Ok good to know. I wasn't sure about it as is was also done in the `settings.component.ts` but as discussed I also removed it there.

**Pull request merged**